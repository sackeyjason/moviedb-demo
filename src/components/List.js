import React from 'react';

function List(props) {
  let items;
  let filter;

  if (props.filter) {
    filter = props.filter.toLowerCase();
    items = props.items.filter(item => {
      return item.name.toLowerCase().indexOf(filter) > -1
    });
  } else {
    items = props.items;
  }
  
  return (
    <ol className="entertainment-list">
      {items.map(item => (
        <li key={item.id}>
          <a href="#!"
            onClick={props.clickHandler.bind(null, item)}>{item.name}</a>
        </li>
      ))}
    </ol>
  )
}

export default List;
