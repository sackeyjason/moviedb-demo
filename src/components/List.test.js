import React from 'react';
import ReactDOM from 'react-dom';
import List from './List';
import renderer from 'react-test-renderer';

const testItems = [
  {name: 'cat', id: 0},
  {name: 'dog', id: 1}
];

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<List
    items={[]}
    filter=""
  />, div);
  ReactDOM.unmountComponentAtNode(div);
});

test('renders item names', () => {
  const list = renderer.create(
    <List
      items={testItems}
      filter=""
      clickHandler={() => {}}
    />
  );
  let tree = list.toJSON();
  expect(tree).toMatchSnapshot();
});

test('filters item names', () => {
  const list = renderer.create(
    <List
      items={testItems}
      filter="cat"
      clickHandler={() => {}}
    />
  );
  let tree = list.toJSON();
  expect(tree).toMatchSnapshot();
});
