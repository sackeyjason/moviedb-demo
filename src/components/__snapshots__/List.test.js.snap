// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`filters item names 1`] = `
<ol
  className="entertainment-list"
>
  <li>
    <a
      href="#!"
      onClick={[Function]}
    >
      cat
    </a>
  </li>
</ol>
`;

exports[`renders item names 1`] = `
<ol
  className="entertainment-list"
>
  <li>
    <a
      href="#!"
      onClick={[Function]}
    >
      cat
    </a>
  </li>
  <li>
    <a
      href="#!"
      onClick={[Function]}
    >
      dog
    </a>
  </li>
</ol>
`;
